MergerService/Utils/TaskUtils.cs

```diff
@@ -13,107 +14,154 @@ public class TaskUtils : ITaskUtils
         }
 
         public void UpdateProgress(string jobId, string taskId, int progress)
         {
-            var content = new FormUrlEncodedContent(new[]
+            using (var activity = this._activitySource.StartActivity("update task progress"))
             {
-                new KeyValuePair<string, string>("percentage", progress.ToString())
-            });
-            Update(jobId, taskId, content);
+                // activity.AddTag("progress", progress);
+
+                using (var content = new StringContent(JsonConvert.SerializeObject(new
+                {
+                    percentage = progress
+                }, this._jsonSerializerSettings)))
+                {
+                    content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+                    Update(jobId, taskId, content);
+                }
+            }
         }
 
         public void UpdateCompletion(string jobId, string taskId)
         {
-            var content = new FormUrlEncodedContent(new[]
+            using (var activity = this._activitySource.StartActivity("update task completed"))
             {
-                new KeyValuePair<string, string>("percentage", "100"),
-                new KeyValuePair<string, string>("status", "completed")
-            });
-            Update(jobId, taskId, content);
+                using (var content = new StringContent(JsonConvert.SerializeObject(new
+                {
+                    percentage = 100,
+                    status = Status.COMPLETED
+                }, this._jsonSerializerSettings)))
+                {
+                    content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+                    Update(jobId, taskId, content);
+                }
+            }
         }
 
         public void UpdateReject(string jobId, string taskId, int attempts, string reason, bool resettable)
         {
-            attempts++;
-
-            // Check if the task should actually fail
-            if (!resettable || attempts == this._maxAttempts)
+            using (var activity = this._activitySource.StartActivity("reject task"))
             {
-                UpdateFailed(jobId, taskId, reason);
-            }
+                attempts++;
+                // activity.AddTag("attempts", attempts);
+                // activity.AddTag("resettable", resettable);
 
-            var content = new FormUrlEncodedContent(new[]
-            {
-                new KeyValuePair<string, string>("attempts", attempts.ToString()),
-                new KeyValuePair<string, string>("reason", reason),
-                new KeyValuePair<string, string>("resettable", resettable.ToString())
-            });
-            Update(jobId, taskId, content);
+                // Check if the task should actually fail
+                if (!resettable || attempts == this._maxAttempts)
+                {
+                    UpdateFailed(jobId, taskId, reason);
```

Comment:
resettable and attempt should be updated on failed as well

Reply:
Added.